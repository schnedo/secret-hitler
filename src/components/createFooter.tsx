import { ComponentType, ReactElement, useState } from "react";
import type { ButtonProps } from "./Button";
import type { LicenseProps } from "./createLicense";

export default function createFooter(
  Button: ComponentType<ButtonProps>,
  License: ComponentType<LicenseProps>,
) {
  return function Footer(): ReactElement {
    const [isOpen, setOpen] = useState(false);

    return (
      <footer>
        <Button onClick={() => setOpen(true)}>License</Button>
        <License open={isOpen} onClose={() => setOpen(false)} />
      </footer>
    );
  };
}
