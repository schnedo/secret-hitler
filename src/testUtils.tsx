export function MockComponent(props: unknown): JSX.Element {
  return <div>{JSON.stringify(props)}</div>;
}
