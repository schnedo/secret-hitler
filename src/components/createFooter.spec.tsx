import { render } from "@testing-library/react";
import { MockComponent } from "../testUtils";
import createFooter from "./createFooter";

it("should render correctly", async () => {
  expect.hasAssertions();

  const Footer = createFooter(MockComponent, MockComponent);
  const { container } = render(<Footer />);

  expect(container).toMatchSnapshot();
});
