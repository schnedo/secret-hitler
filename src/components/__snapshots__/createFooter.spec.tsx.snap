// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <footer>
    <div>
      {"children":"License"}
    </div>
    <div>
      {"open":false}
    </div>
  </footer>
</div>
`;
